### PR TITLE
Updated Typeahead Dropdown

### PR DESCRIPTION
**Pull request by brandenTenbrink**
Closes # https://github.com/influxdata/clockface/issues/761

### Changes

This is a stop gap solution to improve performance for users that have very large dropdown lists.

To utilize the new functionality, set ```largeListSearch = true``` and 
```largeListCeiling = max # of items you want the dropdown list to allow```

The dropdown will force the user to start searching characters to filter the list--items will show once the list.length < the largeListCeiling

### Screenshots
Screen Shot 2022-05-26 at 2 30 14 PM
Screen Shot 2022-05-26 at 2 30 22 PM
Screen Shot 2022-05-26 at 2 30 27 PM
Screen Shot 2022-05-26 at 2 30 32 PM

### Checklist
Check all that apply

- [x] Updated documentation to reflect changes
- [x] Added entry to top of Changelog with link to PR (not issue)
- [x] Tests pass
- [x] Peer reviewed and approved

### CODE REVIEW

File: src/Components/Dropdowns/Composed/TypeAheadDropDown.tsx [hunk 1]
```diff
@@ -90,19 +96,37 @@ export const TypeAheadDropDown: FC<OwnProps> = ({
   const [shownValues, setShownValues] = useState(items)
```
Comment (wdoconnell) [inline src/Components/Dropdowns/Composed/TypeAheadDropDown.tsx:96]: Minor nit, but after this change `shownValues` isn't so semantic anymore and might be confusing. The `shownValues` aren't shown when `largeListValidation` evaluates to true. Maybe we change it to something like `searchResults` or `queryResults` so it's clear that there's a difference between what's being fetched from the list of menu items, versus what's being shown to the user.

Comment (hoorayimhelping) [reply src/Components/Dropdowns/Composed/TypeAheadDropDown.tsx:96]: I think this is a good call

**Review by hoorayimhelping — APPROVED** [review]
couple of minor documentation things (like version and date of release), but overall LGTM

**Review by wdoconnell — APPROVED** [review]

File: src/Components/Dropdowns/Composed/TypeAheadDropDown.tsx (hunk 1 above)
Comment (brandenTenbrink) [reply src/Components/Dropdowns/Composed/TypeAheadDropDown.tsx:96]: Sounds good. I will make this change

File: src/Components/Dropdowns/Composed/TypeAheadDropDown.tsx [hunk 2]
```diff
@@ -70,6 +74,8 @@ export const TypeAheadDropDown: FC<OwnProps> = ({
   testIdSuffix = 'typeAhead',
   selectedOption = null,
   className,
+  largeListSearch = false,
+  largeListCeiling = 0,
```
[the remark this answers is not in the document]
Comment (brandenTenbrink) [reply src/Components/Dropdowns/Composed/TypeAheadDropDown.tsx:78]: Yes, we can default to 15

**Pull request merged**